### DISCUSSION TITLE
Documentation: Persistent cookies

### DISCUSSION THREAD

**Discussion by supervirus** [encode/httpx#2229] [General]
Hi, I saw multiple questions like  #895 or #1056 about persisting cookies in a file.

Maybe it would be helpful to add a small section to the documentation, as there is a quite easy way offered by Python 3 itself....

Persistent cookies can be achieved by passing an [LWPCookieJar](https://docs.python.org/3/library/http.cookiejar.html#http.cookiejar.LWPCookieJar) or [MozillaCookieJar](https://docs.python.org/3/library/http.cookiejar.html#http.cookiejar.MozillaCookieJar) instance as parameter cookies to HTTPX, as it not only accepts a dict, but another CookieJar instance, as well.

Both are implementations of [FileCookieJar](https://docs.python.org/3/library/http.cookiejar.html#http.cookiejar.FileCookieJar), which offers load an save functions on top of the normal CookieJar interface.

```python
from http.cookiejar import LWPCookieJar
import httpx

cookiejar = LWPCookieJar(filename='lwp_cookies.dat')
try:
    cookiejar.load()
except FileNotFoundError:
    pass

session = httpx.Client(cookies=cookiejar)
r = httpx.get('https://httpbin.org/cookies/set?chocolate=chip')

cookiejar.save()
```

```python
from http.cookiejar import MozillaCookieJar
import httpx

cookiejar = MozillaCookieJar(filename='my_cookies.txt')
try:
    cookiejar.load()
except FileNotFoundError:
    pass

session = httpx.Client(cookies=cookiejar)
r = httpx.get('https://httpbin.org/cookies/set?chocolate=chip')

cookiejar.save()
```

**Comment by la-gruge** [comment]
Hi, for your code to work with the LWPCookieJar or MozillaCookieJar, you must use the previously created session to make your GET requests.

from :
```python
r = httpx.get('https://httpbin.org/cookies/set?chocolate=chip')
```
to :
```python
r = session.get('https://httpbin.org/cookies/set?chocolate=chip')
```

**Comment by lahwaacz** [comment]
To add to this: you don't need to keep the cookiejar object around yourself, it is available in `session.cookies.jar`:

```python
session.cookies.jar.save()
```
